### PR TITLE
Feat/hide dropdown on click outside

### LINKED ISSUE #394: [BUG]: When clicking outside of an input the search result should not be visible.
Second if i click outside of an input the search result should not be visible. 

image

_Originally posted by @hermannleboss in https://github.com/globe-and-citizen/cnc-portal/issues/390#issuecomment-2405462281_

### PR DESCRIPTION
**Pull request by dasarathg68**
# Description

Fixes #394 

## Contribution

For your PR please add a commnent to each file edited to explain the changes you made.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

Make sur you run the following commands before submitting your PR:

- /app

```bash
npm run build
npm run test:unit
npm run type-check
npm run lint
npm run format
```

- /backend

```bash
npm run build
npm run test
npm run lint
npm run format
```

- /contract

```bash
npm run test
npm run lint
npm run format
```

- **Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)

### CODE REVIEW

File: app/src/components/forms/CreateOfficerTeam.vue
Comment (dasarathg68) [inline app/src/components/forms/CreateOfficerTeam.vue:1]: Click-outside to hide dropdown menus when a user clicks outside of a specific form element:
1. It creates a reference (`formRef`) to an HTML element.
2. The `handleClickOutside` function checks if a click event occurred outside the referenced element.
3. If the click is outside, it hides the founder and member dropdowns.
4. The `onMounted` hook adds this click event listener to the document when the component is mounted.
5. The `onUnmounted` hook removes the event listener when the component is unmounted, preventing memory leaks.

File: app/src/components/sections/SingleTeamView/forms/AddMemberForm.vue
Comment (dasarathg68) [inline app/src/components/sections/SingleTeamView/forms/AddMemberForm.vue:1]: Same here, but it hides the member menu

File: app/src/components/sections/SingleTeamView/forms/CreateProposalForm.vue
Comment (dasarathg68) [inline app/src/components/sections/SingleTeamView/forms/CreateProposalForm.vue:1]: Same

**Review by hermannleboss — APPROVED** [review]
LGTM

**Pull request merged**